dandi/cli/cmd_service_scripts.py

```diff
@@ -0,0 +1,103 @@
+from difflib import unified_diff
+import json
+from pathlib import PurePosixPath
+
+import click
+from dandischema.consts import DANDI_SCHEMA_VERSION
+from packaging.version import Version
+
+from .base import map_to_click_exceptions
+from .. import lgr
+from ..dandiapi import RemoteBlobAsset
+from ..dandiarchive import parse_dandi_url
+from ..exceptions import NotFoundError
+
+
+@click.group()
+def service_scripts() -> None:
+    """Various utility operations"""
+    pass
+
+
+@service_scripts.command()
+@click.option("--diff", is_flag=True, help="Show diffs of old & new metadata")
+@click.option(
+    "--when",
+    type=click.Choice(["newer-schema-version", "always"]),
+    default="newer-schema-version",
+    help="When to re-extract an asset's metadata",
+    show_default=True,
+)
+@click.argument("url")
+@map_to_click_exceptions
+def reextract_metadata(url: str, diff: bool, when: str) -> None:
+    """
+    Recompute & update the metadata for NWB assets on a remote server.
+
+    Running this command requires the fsspec library to be installed with the
+    `http` extra (e.g., `pip install "fsspec[http]"`).
+    """
+    from ..metadata import nwb2asset  # Avoid heavy import at top level
+
+    parsed_url = parse_dandi_url(url)
+    if parsed_url.version_id not in (None, "draft"):
+        raise click.UsageError("URL cannot point to a published Dandiset")
+    parsed_url.version_id = "draft"
```

Comment:
here I fell into the trap of the fact that whenever a dandiset has a released version, then non-versioned interactions go with the latest released version. That caused my confusion as depicted in another comment/recommendation to adjust logging.

Let's do it more explicit, and also guard against usage of URLs without dandiset specified (e.g. `https://dandiarchive.org/assets/838bab7b-9ab4-4d66-97b3-898a367c9c7e/`):
```suggestion
    if not parsed_url.dandiset_id:
        raise click.UsageError("URL must point to an asset within a Dandiset.")
    if parsed_url.version_id != "draft":
        raise click.UsageError(
            "URL must point explicitly to a draft version of a Dandiset. Got %s",
            parsed_url
        )
```